### PR TITLE
fix: add linking of both users

### PR DESCRIPTION
**Pull request by markusahlstrand**
Added a call so that the linked user gets a `linked_with` field set in the profile.

### CODE REVIEW

File: src/routes/tsoa/users-mgmt.ts [hunk 1]
```diff
@@ -214,7 +215,7 @@ export class UsersMgmtController extends Controller {
     const currentDbUser = await db
       .selectFrom("users")
       .where("users.tenantId", "=", tenantId)
-      .where("users.id", "=", request.ctx.state.sub)
+      .where("users.id", "=", userId)
```
Comment (danielgent) [inline src/routes/tsoa/users-mgmt.ts:218]: :+1:

File: src/routes/tsoa/users-mgmt.ts [hunk 2]
```diff
@@ -241,6 +242,18 @@ export class UsersMgmtController extends Controller {
...
+      email: linkedDbUser.email,
+      linkWithEmail: currentDbUser.email,
+    });
+
+    // Link the parent account
+    return currentUser.linkWithUser.mutate({
+      tenantId,
+      email: currentDbUser.email,
+      linkWithEmail: linkedDbUser.email,
+    });
```
Comment (danielgent) [inline src/routes/tsoa/users-mgmt.ts:257]: is there any way to unlink accounts?

I'm reading through the Auth0 docs to understand what they do under-the-hood

I think you've spent enough time linking accounts though so I'm happy to trust you :smile:

File: src/routes/tsoa/users-mgmt.ts [hunk 3]
```diff
@@ -241,6 +242,18 @@ export class UsersMgmtController extends Controller {
       getId(tenantId, linkedDbUser.email),
     );
 
-    return currentUser.getProfile.query();
+    // Link the account child account
```
Comment (danielgent) [inline src/routes/tsoa/users-mgmt.ts:245]:
```suggestion
    // Link the child account
```

**Review by danielgent — APPROVED** [review]

File: src/routes/tsoa/users-mgmt.ts (hunk 2 above)
Comment (markusahlstrand) [reply src/routes/tsoa/users-mgmt.ts:257]: There's an endpoint to do it in auth0 but we haven't implemented that yet :)

**Pull request merged**